src/lightning/fabric/strategies/xla.py

```diff
@@ -211,7 +217,12 @@ def broadcast(self, obj: TBroadcast, src: int = 0) -> TBroadcast:
         if is_tensor:
             if obj.dim() == 0:
                 obj = obj.unsqueeze(0)
-            if obj.device.type != "xla":
+            original_device = obj.device
+            if original_device.type != "xla":
+                # XLA distributed requires that the data is on the XLA device
+                # TODO: this might OOM if a CPU object is broadcast that wouldn't fit on XLA's memory. a workaround
+                # for that would be to use regular torch distributed for CPU objects with gloo. however, that would
+                # have other challenges such as having to initialize it and the lack of bfloat16 support for gloo
```

Comment:
It's good to mention it, but I think it is not that big of a deal or surprising. For cuda it is the same, the tensor needs to be on the cuda device for collectives like reduce, gather etc. to work.